packages/arb-token-bridge-ui/src/token-bridge-sdk/Erc20TeleportStarter.ts

```diff
@@ -0,0 +1,192 @@
+import { Erc20L1L3Bridger, getL2Network } from '@arbitrum/sdk'
+import { ERC20__factory } from '@arbitrum/sdk/dist/lib/abi/factories/ERC20__factory'
+import { constants } from 'ethers'
+import {
+  ApproveNativeCurrencyProps,
+  ApproveTokenProps,
+  BridgeTransferStarter,
+  BridgeTransferStarterProps,
+  RequiresNativeCurrencyApprovalProps,
+  RequiresTokenApprovalProps,
+  TransferEstimateGas,
+  TransferProps,
+  TransferType
+} from './BridgeTransferStarter'
+import { fetchErc20Allowance } from '../util/TokenUtils'
+import { getAddressFromSigner } from './utils'
+import { getL2ConfigForTeleport } from './teleport'
+
+export class Erc20TeleportStarter extends BridgeTransferStarter {
+  public transferType: TransferType = 'erc20_teleport'
+
+  constructor(props: BridgeTransferStarterProps) {
+    super(props)
+
+    if (!this.sourceChainErc20Address) {
+      throw Error('Erc20 token address not found')
+    }
+  }
+
+  public async requiresNativeCurrencyApproval({
+    amount,
+    signer
+  }: RequiresNativeCurrencyApprovalProps) {
+    return false
+  }
+
+  public async approveNativeCurrency({
```

Comment:
If we're skipping chains that have custom gas token, then this should not do anything

Reply:
Removed the code